packages/std/src/math.rs

```diff
@@ -5,26 +5,43 @@ use std::{fmt, ops};
 
 use crate::errors::{generic_err, underflow, StdError, StdResult};
 
-/// Decimal9 represents a fixed-point decimal value with 9 fractional digits.
-/// That is Decimal9(1_000_000_000) == 1
+/// A fixed-point decimal value with 18 fractional digits, i.e. Decimal(1_000_000_000_000_000_000) == 1.0
 #[derive(Serialize, Deserialize, Copy, Clone, Debug, PartialEq, JsonSchema)]
-pub struct Decimal9(pub u64);
+pub struct Decimal(Uint128);
```

Comment:
This is a nice short-cut, but it means we encode 1.234 as "1234000000000000000" not "1.23400..." I think it would be ideal to use the second, so this info (18 digits) doesn't need to be known out-of-bounds. That said, I will check how the cosmos-sdk encodes it, to make sure we are compatible.

Reply:
```go
func TestDemoDec(t *testing.T) {
	d, err := NewDecFromStr("1234.678000")
	require.NoError(t, err)
	fmt.Println(d.String())
	f := NewDec(765430)
	fmt.Println(f.String())

	bz, err := json.Marshal(d)
	require.NoError(t, err)
	fmt.Println(string(bz))}
```

returns:

```
=== RUN   TestDemoDec
1234.678000000000000000
765430.000000000000000000
"1234.678000000000000000"
--- PASS: TestDemoDec (0.00s)
```

Reply:
For all serde serialization you need to know the type in order to interprete the serialization.

But we can make it a decimal string for debugability. It just needs to error when receiving a decimal with more digits than supported.

The type needs a string parser anyways (sooner or later)

Reply:
Yes, we need to define the type properly. But if someone sends "12.3456" it should work properly. (even though the sdk seems to keep trailing zeros by default)